src/main/java/de/hysky/skyblocker/skyblock/chocolatefactory/EggFinder.java

```diff
@@ -39,18 +40,39 @@ public class EggFinder {
 	private static final Pattern eggFoundPattern = Pattern.compile("^(?:HOPPITY'S HUNT You found a Chocolate|You have already collected this Chocolate) (Breakfast|Lunch|Dinner)");
 	private static final Pattern newEggPattern = Pattern.compile("^HOPPITY'S HUNT A Chocolate (Breakfast|Lunch|Dinner) Egg has appeared!$");
 	private static final Logger logger = LoggerFactory.getLogger("Skyblocker Egg Finder");
+	//This is most likely unnecessary with the addition of the location change packet, but it works fine and might be doing something so might as well keep it
 	private static final LinkedList<ArmorStandEntity> armorStandQueue = new LinkedList<>();
+	/**
+	 * The locations that the egg finder should work while the player is in.
+	 */
 	private static final Location[] possibleLocations = {Location.CRIMSON_ISLE, Location.CRYSTAL_HOLLOWS, Location.DUNGEON_HUB, Location.DWARVEN_MINES, Location.HUB, Location.THE_END, Location.THE_PARK, Location.GOLD_MINE, Location.DEEP_CAVERNS, Location.SPIDERS_DEN, Location.THE_FARMING_ISLAND};
+	/**
+	 * Whether the player is in a location where the egg finder should work.
+	 * This is set to false upon world change and will be checked with the location change event afterward.
+	 */
 	private static boolean isLocationCorrect = false;
 
-	private EggFinder() {
-	}
+	private EggFinder() {}
 
 	public static void init() {
-		ClientPlayConnectionEvents.JOIN.register((ignored, ignored2, ignored3) -> invalidateState());
+		ClientPlayConnectionEvents.JOIN.register((ignored, ignored2, ignored3) -> isLocationCorrect = false);
 		SkyblockEvents.LOCATION_CHANGE.register(EggFinder::handleLocationChange);
 		ClientReceiveMessageEvents.GAME.register(EggFinder::onChatMessage);
 		WorldRenderEvents.AFTER_TRANSLUCENT.register(EggFinder::renderWaypoints);
+		ClientTickEvents.END_CLIENT_TICK.register(client -> {
+			if (!SkyblockerConfigManager.get().helpers.chocolateFactory.enableEggFinder || client.player == null) return;
+			for (EggType type : EggType.entries) {
```

Comment:
there should be some more checks to ensure that this isn't constantly creating an iterator, such as being in skyblock, correct location & time